src/nashpy/algorithms/lemke_howson.py

```diff
@@ -2,73 +2,16 @@
 import warnings
 from itertools import cycle
 
-import numpy as np
 import numpy.typing as npt
-from typing import Tuple, Set, Iterable
-from nashpy.integer_pivoting import (
-    make_tableau,
-    non_basic_variables,
-    pivot_tableau,
-)
-
-
-def shift_tableau(tableau: npt.NDArray, shape: Tuple[int, ...]) -> npt.NDArray:
-    """
-    Shift a tableau to ensure labels of pairs of tableaux coincide
-
-    Parameters
-    ----------
-    tableau : array
-        a tableau corresponding to a vertex of a polytope.
-    shape : tuple
-        the required shape of the tableau
-
-    Returns
-    -------
-    array
-        The shifted tableau
-    """
-    return np.append(
-        np.roll(tableau[:, :-1], shape[0], axis=1),
-        np.ones((shape[0], 1)),
-        axis=1,
-    )
-
-
-def tableau_to_strategy(
-    tableau: npt.NDArray, basic_labels: Set[int], strategy_labels: Iterable
-) -> npt.NDArray:
-    """
-    Return a strategy vector from a tableau
-
-    Parameters
-    ----------
-    tableau : array
-        a tableau corresponding to a vertex of a polytope.
-    basic_labels : set
-        the set of basic labels.
-    strategy_labels : Iterable
-        the set of labels that correspond to strategies.
-
-    Returns
-    -------
-    array
-        A strategy.
-    """
-    vertex = []
-    for column in strategy_labels:
-        if column in basic_labels:
-            for i, row in enumerate(tableau[:, column]):
-                if row != 0:
-                    vertex.append(tableau[i, -1] / row)
-        else:
-            vertex.append(0)
-    strategy = np.array(vertex)
-    return strategy / sum(strategy)
+from typing import Tuple
+from nashpy.linalg import TableauBuilder
 
 
 def lemke_howson(
-    A: npt.NDArray, B: npt.NDArray, initial_dropped_label: int = 0
+    A: npt.NDArray,
+    B: npt.NDArray,
+    initial_dropped_label: int = 0,
+    algorithm: str = "basic",
```

Comment:
What do you think about changing this to:

```suggestion
    lexicographic: bool = False,
```

I'm wondering however if perhaps it might be worth just having lexicographic as the default -- in which case perhaps there's no need for the non lexicographic Tableau at all. What do you think?

Reply:
I think there are a few questions packed into one here, will try to answer them all seperately

1. should non-lexiographic tableau exist at all:
I understand that part of the purpose of the library is being a teaching tool. From that perspective I think it really makes sense to offer both, as the lexiographic class sort of highlights that these are the few places you'd adapt lemke howson in order to handle degenerate matrices. At least for me just understanding the basic algorithm was already complex enough, and the theory around lexiographic sorting was a very cryptic layer on top of that again. So offering lex sort as an extension to the normal tableau makes the differences very explicit.
2. what should the default/options be when calling `Game::lemke_howson_enumeration``:
I took the conservative approach of keeping the behavior exactly the same as today. I probably think you should decide what the most sensible defaults and options are. I randomly stumbled into nash equilibriums and most of my cases were degenerate games. So I would maybe think that given the performance is quite similar but lex-sorting more versatile, I would lean towards at least default being lex sorting.
3. String algorithm, vs bool lexiographic option
I drew inspiration from https://docs.scipy.org/doc/scipy/tutorial/optimize.html when going for string. Still I agree bool makes sense if there are just two different implementations. If you think that is the case, I'd be happy to make that change, though potentially I see there might be more algorithms like:
* until I managed to solve the lex sorting bug I considered experimenting with the cycle-breaking approach of picking a different row to pivot each time we had the same basic variables and same label to drop
* could there be other pivoting algorithms (potentially with or without lex sorting). What we do today maybe emphasizes numerical and integer stability, but could run into infs and nans on large matrices. Potentially there could be an algorithm that handles those types of matrices better.

Reply:
Thank you very much for your thoughtful response.

> should non-lexiographic tableau exist at all:
> I understand that part of the purpose of the library is being a teaching tool. From that perspective I think it really makes sense to offer both, as the lexiographic class sort of highlights that these are the few places you'd adapt lemke howson in order to handle degenerate matrices. At least for me just understanding the basic algorithm was already complex enough, and the theory around lexiographic sorting was a very cryptic layer on top of that again. So offering lex sort as an extension to the normal tableau makes the differences very explicit.

I agree. This will also help when I refactor the documentation on LH.

> what should the default/options be when calling `Game::lemke_howson_enumeration``:
I took the conservative approach of keeping the behavior exactly the same as today. I probably think you should decide what the most sensible defaults and options are. 

I appreciate the way you've put this PR together greatly :)

> I randomly stumbled into nash equilibriums and most of my cases were degenerate games. So I would maybe think that given the performance is quite similar but lex-sorting more versatile, I would lean towards at least default being lex sorting.

Let's do it! If you could make the change to make lex the default that'd be great.

> String algorithm, vs bool lexiographic option
I drew inspiration from https://docs.scipy.org/doc/scipy/tutorial/optimize.html when going for string. Still I agree bool makes sense if there are just two different implementations. If you think that is the case, I'd be happy to make that change, though potentially I see there might be more algorithms like:
until I managed to solve the lex sorting bug I considered experimenting with the cycle-breaking approach of picking a different row to pivot each time we had the same basic variables and same label to drop
could there be other pivoting algorithms (potentially with or without lex sorting). What we do today maybe emphasizes numerical and integer stability, but could run into infs and nans on large matrices. Potentially there could be an algorithm that handles those types of matrices better.

Let's go for bool, I think we're on the same page here. I don't want to over optimise for future changes though so let's just go with what's nice now.

Reply:
Switched to lexiographic boolean and default is on